### PR TITLE
first pass at opening issue in browser

### PR DESCRIPTION
**Pull request by yyyc514**

### CODE REVIEW

**Comment by yyyc514** [comment]
screen shot 2015-03-26 at 10 17 37 pm

First pass at this.  Currently only loads a single page of issues (same as the issue fetch for commits).  Need some feedback on how to choose a remote.  I changed last=True so it works for GitSavvy... but really there are two big use cases I can think of:

- remote with issues is origin (you're a direct contributor)
- remote with issues is upstream or some other (you're an indirect contributor via github)

I think we need a nice way to handle both without having to make a choice each time.

**Comment by yyyc514** [comment]
Related #14

**Comment by yyyc514** [comment]
And this doesn't support timezones properly.  Dates from github are UTC I'm guessing?  Not sure.

**Comment by divmain** [comment]
Remote-selection should be drastically simplified as of 93cf346302f7c2baad87604fbe2b80048acd2a81, so take a look there.  I'm not sure what timezone GitHub uses in their API responses, haven't had to deal with that yet.  Is there a timezone included in the timestamps themselves, or as header information somewhere in the payload?

With `git: log` quick panel, we have paging set at something like 5000 commits.  You could do something similar here, where you add an entry at the end to page forward to the next commits.  It isn't perfect, but it is still a big improvement on opening browser, clicking issues, searching for the right issue.

**Comment by yyyc514** [comment]
> Remote-selection should be drastically simplified as of 93cf346, so take a look there.

Yep, I'll go back and refactor this with your recent changes in mind.

> Is there a timezone included in the timestamps themselves, or as header information somewhere in the payload?

Timestamps appear to be UTC I think... I'll have to look again for some type of header - I didn't see one.

> With git: log quick panel, we have paging set at something like 5000 commits. You could do something similar here, where you add an entry at the end to page forward to the next commits.

It's 6000. Funny number.  Ok, that makes sense as a way to get this workable and out the door.  I'll see about doing that.

**Comment by yyyc514** [comment]
Thoughts?

File: core/git_command.py [hunk 1]
```diff
@@ -31,6 +31,22 @@ class GitSavvyError(Exception):
...
 
+class IsolatedGitCommand(GitCommand):
+    """
+    Allow accessing GitCommand outside the scope of a WindowCommand or
+    TextCommand.  You must manually pass in a valid repository path.
+    """
+    def __init__(self, repo_path=None):
+        self._repo_path = repo_path
+
+    def git(self, *args):
```
Comment (divmain) [inline core/git_command.py]: This method does nothing.

File: core/git_command.py [hunk 2]
```diff
@@ -31,6 +31,22 @@ class GitSavvyError(Exception):
     pass
 
 
+class IsolatedGitCommand(GitCommand):
+    """
+    Allow accessing GitCommand outside the scope of a WindowCommand or
+    TextCommand.  You must manually pass in a valid repository path.
+    """
+    def __init__(self, repo_path=None):
+        self._repo_path = repo_path
```
Comment (divmain) [inline core/git_command.py]: You can just say `self.repo_path = repo_path` and remove the `repo_path` property method.

File: core/git_command.py [hunk 3]
```diff
@@ -31,6 +31,22 @@ class GitSavvyError(Exception):
     pass
 
 
+class IsolatedGitCommand(GitCommand):
```
Comment (divmain) [inline core/git_command.py]: Not sure this is the abstraction I'd like to go with, so will think on this a bit before moving forward.

File: github/github.py [hunk 4]
```diff
@@ -80,6 +93,11 @@ def get_api_fqdn(github_repo):
 
 
 def query_github(api_url_template, github_repo):
+    resp = raw_query_github(api_url_template, github_repo)
+    return resp.payload
+
+
+def raw_query_github(api_url_template, github_repo):
```
Comment (divmain) [inline github/github.py]: This is an unnecessary change.

**Comment by divmain** [comment]
As I mentioned, not sure `IsolatedGitCommand` is how I'd like this to work.  I'll think on this a bit more and get back to you.  Also, please add docstrings all around.

File: github/github.py (hunk 4 above)
Comment (yyyc514) [inline github/github.py]: Do you want the raw response returned everywhere and all code that uses web requests to be modified then?

File: core/git_command.py (hunk 3 above)
Comment (yyyc514) [inline core/git_command.py]: Yeah it's a little different, but I needed git outside of a command/window so I didn't need the rest of the baggage... really what should be abstracted is running commands in general then git on top of that, then GitCommand as used by window/view.

File: core/git_command.py (hunk 1 above)
Comment (yyyc514) [inline core/git_command.py]: I thought it was allowing me to call git with fewer options but now seeing that the calls are both *args that doesn't seem to be the case.

File: core/git_command.py (hunk 2 above)
Comment (yyyc514) [inline core/git_command.py]: Are you sure?  Because this is a subclass of GitCommand I needed to override it's behavior for retrieving repo_path... does just setting a static value kill the property method from the parent class?

**Comment by yyyc514** [comment]
>  Also, please add docstrings all around.

Where specifically?  In reviewing a git diff not much was added really.  I added a doctoring to the UTC method... but most new things have them, unless I'm missing something?

**Comment by divmain** [comment]
I closed the ticket but failed to close this PR.  From the issue:

> Have decided this is out of scope for GitSavvy. Could be implemented as a GitSavvy plugin. In my case, I've discovered http://overv.io for issue management, and I doubt anything I put in GitSavvy (aside from the light integration that already exists) could provide a better issue management experience.
